include/gridtools/common/int_vector.hpp

```diff
@@ -0,0 +1,134 @@
+
+namespace gridtools {
+    GT_TARGET_NAMESPACE {}
+
+    namespace int_vector {
+        GT_TARGET_NAMESPACE {
+
+            namespace int_vector_detail {
+                template <class I, class Enable = void, class Key = meta::first<I>, class Type = meta::second<I>>
+                struct add_f {
+                    template <class First, class Second>
+                    GT_TARGET GT_FORCE_INLINE GT_TARGET_CONSTEXPR Type operator()(
+                        First &&first, Second &&second) const {
+                        return at_key_with_default<Key, std::integral_constant<Type, 0>>(wstd::forward<First>(first)) +
+                               at_key_with_default<Key, std::integral_constant<Type, 0>>(wstd::forward<Second>(second));
+                    }
+                };
+                template <class I>
+                struct add_f<I, std::enable_if_t<is_integral_constant<meta::second<I>>::value>> {
+                    template <class First, class Second>
+                    GT_TARGET GT_FORCE_INLINE GT_TARGET_CONSTEXPR meta::second<I> operator()(
+                        First &&, Second &&) const {
+                        return {};
+                    }
+                };
+
+                // TODO or the following?
+
+                // template <class I, class Key = meta::first<I>, class Type = meta::second<I>>
+                // struct add_f {
+                //     template <class First, class Second>
+                //     Type operator()(First &&first, Second &&second) const {
+                //         if constexpr (is_integral_constant<Type>::value) {
+                //             return {};
+                //         } else {
+                //             return at_key_with_default<Key, std::integral_constant<Type, 0>>(first) +
+                //                    at_key_with_default<Key, std::integral_constant<Type, 0>>(second);
+                //         }
+                //     }
+                // };
+
+                template <class First, class Second>
+                GT_TARGET GT_FORCE_INLINE GT_TARGET_CONSTEXPR auto plus(First &&first, Second &&second) {
+                    using merged_meta_map_t = meta::mp_make<int_vector_impl_::merger_t,
+                        meta::concat<hymap::to_meta_map<First>, hymap::to_meta_map<Second>>>;
+                    using keys_t = meta::transform<meta::first, merged_meta_map_t>;
+                    using generators = meta::transform<add_f, merged_meta_map_t>;
+                    return tuple_util::generate<generators, hymap::from_meta_map<merged_meta_map_t>>(
+                        std::forward<First>(first), std::forward<Second>(second));
+                }
+
+                template <class Vec, class Scalar>
+                GT_TARGET GT_FORCE_INLINE GT_TARGET_CONSTEXPR auto multiply(Vec &&vec, Scalar &&scalar) {
+                    return tuple_util::transform([&scalar](auto &&v) { return v * scalar; }, std::forward<Vec>(vec));
```

Comment:
`tuple_util::transform` is an alias of `tuple_util::host::transform`. Is is not what you want here